### DISCUSSION TITLE
TYPE ERROR - > Cannot destructure property 'prevSubject' of 'options' as it is undefined.

### DISCUSSION THREAD

**Discussion by Louisrivera5** [cypress-io/cypress#28486] [Questions and Help]
The following error originated from your test code, not from Cypress.
  > Cannot destructure property 'prevSubject' of 'options' as it is undefined.

When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure.

Im getting this in every spec file. I already downgrade my cypress version and it is still doing the same thing, I even write a simple test but it still produce the same error. 

![Cypress Error _ TypeError 12 09 23](https://github.com/cypress-io/cypress/assets/138297068/6e1f3ac3-234b-48ac-96fc-564d4c539cde)

I hope you can help me.

**Comment by adamalston** [comment]
Can you share the code that is throwing these errors? Can you share more about this statement as well?

> We dynamically generated a new test to display this failure.

**Reply by Louisrivera5** [reply]
Cypress.Commands.add("SelectProduct"), (productName) => {
    cy.get('h4.card-title').each(($el, index, $list) => {
        if($el.text().includes(productName))
        {
                cy.get('button.btn.btn-info').eq(index).click()
        }
    })
};


We dynamically generated a new test to display this failure. 
 - I just copy paste this from Cypress error. 

This the sample test case. 
/// <reference types="Cypress" />
/// <reference types="cypress-iframe" />


//cypress - Spec

describe ('Framework testing 1', function() 
{
  before(function() {
      //Runs before all test run in the IT block      
      cy.fixture('example').then(function(data)
            { 
this.data=data;
            })
          })
// Frameworks ==> fixtures and concept of this.data

it('Framework1', function() {

        cy.visit('https://rahulshettyacademy.com/angularpractice/')

cy.get('input[name="name"]:nth-child(2)').type(this.data.name)
cy.get('select').select(this.data.gender)
cy.get(':nth-child(4) > .ng-untouched').should('have.value',this.data.name)
cy.get('input[name="name"]:nth-child(2)').should('have.attr','minlenght',2)
cy.get('#inlineRadio3').should('be.disabled')
cy.get(':nth-child(2) > .nav-link').click()
cy.selectProduct('Blackberry')
cy.selectProduct('Nokia Edge')

**Comment by em-twines** [comment]
This is such a silly mistake on my part, but in case it helps anyone else -- I had an extra line -- Cypress.Commands.add() -- in my code (in commands,js). Deleting that solved this issue for me. Good luck on your own situations!
